e2e/cluster.go

```diff
@@ -6,30 +6,64 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
 )
 
 func setupCluster(ctx context.Context, cloud *azureClient, location, resourceGroupName, clusterName string) error {
-	aksCluster, err := cloud.aksClient.Get(ctx, resourceGroupName, clusterName, nil)
-	if err != nil {
-		return fmt.Errorf("failed to get aks cluster: %q", err)
-	}
+	var needNewCluster bool
 
-	rgExistence, err := cloud.resourceGroupClient.CheckExistence(ctx, agentbakerTestResourceGroupName, nil)
+	rgExistence, err := cloud.resourceGroupClient.CheckExistence(ctx, resourceGroupName, nil)
 	if err != nil {
-		return fmt.Errorf("failed to get MC RG: %q", err)
+		return fmt.Errorf("failed to get AB E2E RG %q: %q", resourceGroupName, err)
 	}
 
-	if !rgExistence.Success || aksCluster.Properties == nil || aksCluster.Properties.ProvisioningState == nil || *aksCluster.Properties.ProvisioningState == "Failed" {
-		poller, err := cloud.aksClient.BeginDelete(ctx, resourceGroupName, clusterName, nil)
+	if !rgExistence.Success {
+		needNewCluster = true
+		_, err := cloud.resourceGroupClient.CreateOrUpdate(
+			ctx,
+			resourceGroupName,
+			armresources.ResourceGroup{
+				Location: to.Ptr(e2eTestLocation),
+				Name:     to.Ptr(resourceGroupName),
+			},
+			nil)
+
 		if err != nil {
-			return fmt.Errorf("failed to start aks cluster deletion: %q", err)
+			return fmt.Errorf("failed to create AB E2E RG %q: %q", resourceGroupName, err)
 		}
+	}
 
-		_, err = poller.PollUntilDone(ctx, nil)
+	if !needNewCluster {
```

Comment:
nit - bit confusing to check for needNewCluster then fetch the cluster...suppose the condition names could be more unique but not a big deal

Reply:
agreed, refactored this to make things a bit more understandable